catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfRenderPlugin.java

```diff
@@ -0,0 +1,472 @@
+
+    private static final double DEFAULT_OVERVIEW_MAX_SIDE_LENGTH = 1024.0;
+
+    private double overviewMaxSideLength = DEFAULT_OVERVIEW_MAX_SIDE_LENGTH;
+
+    private static final int DEFAULT_MAX_NITF_SIZE_MB = 120;
+
+    private int maxNitfSizeMB = DEFAULT_MAX_NITF_SIZE_MB;
+
+    private boolean storeOriginalImage = true;
+
+    private boolean createOverview = true;
+
+    @Override
+    public ProcessRequest<ProcessCreateItem> processCreate(
+            ProcessRequest<ProcessCreateItem> processRequest) throws PluginExecutionException {
+        return processRequest(processRequest);
+    }
+
+    @Override
+    public ProcessRequest<ProcessUpdateItem> processUpdate(
+            ProcessRequest<ProcessUpdateItem> processRequest) throws PluginExecutionException {
+        return processRequest(processRequest);
+    }
+
+    @Override
+    public ProcessRequest<ProcessDeleteItem> processDelete(
+            ProcessRequest<ProcessDeleteItem> processRequest) throws PluginExecutionException {
+        if (processRequest == null) {
+            throw new PluginExecutionException(NULL_INPUT_MSG);
+        }
+
+        // no work to do
+        return processRequest;
+    }
+
+    private static boolean isNitfMimeType(String rawMimeType) {
+        try {
+            return NITF_MIME_TYPE.match(rawMimeType);
+        } catch (MimeTypeParseException e) {
+            LOGGER.debug("unable to compare mime types: {} vs {}. NITF image will not be processed.",
+                    NITF_MIME_TYPE,
+                    rawMimeType);
+        }
+
+        return false;
+    }
+
+    private <T extends ProcessResourceItem> ProcessRequest<T> processRequest(
+            ProcessRequest<T> input) throws PluginExecutionException {
+        if (input == null) {
+            throw new PluginExecutionException(NULL_INPUT_MSG);
+        }
+
+        List<T> processResourceItems = input.getProcessItems();
+
+        if (processResourceItems != null) {
+            List<T> newProcessResourceItems = new ArrayList<>();
+
+            for (T processResourceItem : processResourceItems) {
+                ProcessResource processResource = processResourceItem.getProcessResource();
+                Metacard metacard = processResourceItem.getMetacard();
+
+                if (processResource == null || !isNitfMimeType(processResource.getMimeType())) {
+                    // skip processing
+                    break;
+                }
+
+                final long megabyte = 1024L * 1024L;
+                if (processResource.getSize() / megabyte > maxNitfSizeMB) {
+                    LOGGER.debug("Skipping large ({} MB) process resource: name={}.",
+                            processResource.getSize() / megabyte,
+                            processResource.getName());
+                    // skip processing
+                    break;
+                }
+
+                try {
+                    final BufferedImage renderedImage = render(processResource);
```

Comment:
Instread of using just NitfRenderer.renderToClosestDateModel for all rendering, I think you should replicate the original logic of using NitfRenderer.render and NitfRenderer.renderToClosestDateModel. Otherwise, you would need to do a lot testing with NITF files to make sure thumbnail and overview images are rendered as they were before within the UI.